src/mpi/messages.rs

```diff
@@ -162,6 +186,18 @@ impl Agent {
             .unwrap()
     }
 
+    pub fn push_leg(&mut self, dep_time: Option<u32>, trav_time: Option<u32>, route: Vec<u64>) {
+        self.plan.as_mut().unwrap().legs.push(Leg {
+            mode: "car".to_string(),
+            dep_time,
+            trav_time,
+            route: Some(Route::NetworkRoute(NetworkRoute {
+                vehicle_id: 0, //TODO
```

Comment:
For starters this could be the same id as the agent. Then at least each vehicle has a unique id.